### PR TITLE
pythonPackages.scrapy: 2.3.0 -> 2.4.0

### PR DESCRIPTION
**Pull request by marsam**
###### Motivation for this change

https://github.com/scrapy/scrapy/releases/tag/2.4.0

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by marsam** [comment]
@GrahamcOfBorg build python3Packages.scrapy

**Comment by marsam** [comment]
@GrahamcOfBorg build python3Packages.scrapy

File: pkgs/development/python-modules/scrapy/default.nix [hunk 1]
```diff
@@ -63,19 +62,28 @@ buildPythonPackage rec {
...
-    pytest --ignore=tests/test_linkextractors_deprecated.py --ignore=tests/test_proxy_connect.py --ignore=tests/test_utils_display.py --deselect tests/test_crawl.py::CrawlTestCase::test_retry_dns_error ${lib.optionalString stdenv.isDarwin "--deselect tests/test_utils_iterators.py::LxmlXmliterTestCase::test_xmliter_encoding"}
   '';
 
+  pytestFlagsArray = [
+    "--ignore=tests/test_proxy_connect.py"
+    "--ignore=tests/test_utils_display.py"
+    "--ignore=tests/test_command_check.py"
+    "--deselect=tests/test_feedexport.py::FTPFeedStorageTest"
+    "--deselect=tests/test_request_left.py::TestCatching::test_noconnect"
+    "--deselect=tests/test_crawl.py::CrawlTestCase::test_retry_dns_error"
```
Comment (jonringer) [inline pkgs/development/python-modules/scrapy/default.nix:76]: for individual tests, you can do 
```suggestion
    disabledTests = [ "test_retry_dns_error" ];
```
easier to read than `--deselect={path}::{Module}::{method}`

Comment (marsam) [reply pkgs/development/python-modules/scrapy/default.nix]: Fixed, thanks!

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 100283 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>8 packages failed to build:</summary>  <ul>    <li>python37Packages.scrapy</li>    <li>python37Packages.scrapy-deltafetch</li>    <li>python37Packages.scrapy-fake-useragent</li>    <li>python37Packages.scrapy-splash</li>    <li>python38Packages.scrapy</li>    <li>python38Packages.scrapy-deltafetch</li>    <li>python38Packages.scrapy-fake-useragent</li>    <li>python38Packages.scrapy-splash</li>  </ul></details>

**Pull request merged**